File: py/9.palindrome_number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        input = x
        check = 0
        while input > 0:
            check = check * 10 + input % 10
            input = input // 10
        return check == x

File: cpp/9.palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        int input = x;
        long int check = 0;
        while (input > 0) {
            check = check * 10 + input % 10;
            input /= 10;
        }
        return check == x;
    }
};
